### ISSUE TITLE
2.1.1 expand_keywords crashes when given empty tags

### ISSUE THREAD

**Issue by CajunAvenger** [twanvl/MagicSetEditor2#90]
a. Using the Bold, Italic, or Symbols formatting button (or dropdown from the Format menu) in a card's blank rules text crashes MSE,
b. On a card that already has text, clicking the button once won't crash MSE, but clicking it a second time will.
c. Using the keyboard shortcuts for these three formatting options doesn't crash MSE, but after using one, clicking its button equivalent will crash MSE.
d. Additionally, if you use the keyboard shortcuts and start typing, you get constant warnings from combined_editor, though MSE seems to continue working fine. The set will save, and on reopening the errors won't recur unless you do this again. [Warning images](https://cdn.discordapp.com/attachments/354683158526033921/749763152811524166/unknown.png)

Simply highlighting the text to format and using the keyboard shortcuts or buttons causes no issues at all, it all happens in toggling the 'always on' mode. D is only happening on fields using combined_editor. A, B, and C seem to occur on all frames regardless of combined_editor usage, but on those that do, this isn't occurring in the flavor text parts of those fields. It does not appear to be `<i-auto>` that is correcting the problem, as the M15 Saga frame uses this in its first textbox and the error still occurs, so possibly due to being the second field of a combined_editor?

**Comment by CajunAvenger** [comment]
Checked the previous versions, this was introduced in 2.0.2-rc5.

**Comment by CajunAvenger** [comment]
Okay after tinkering with the other error, I've been able to track this down a little better. This crash occurs from an empty tags such as `<b></b>` being given to expand_keywords. It seems to occur if the tag makes up the entire string (causing the crash in A) or if the string ends in two tags, including `<b>text</b><b></b>` (causing the B and C crashes).

I've made an adjustment to the text filter that removes the blank tags before running expand_keywords in the text filter, then puts them back in afterwords, and everything is now functioning normally again.
